src/main/java/com/teamfive/disscard/service/CardServiceStub.java

```diff
@@ -20,6 +21,10 @@ public CardServiceStub(ICardDAO cardDAO) {
         this.cardDAO = cardDAO;
     }
 
+    private boolean isValidCard(Card card) {
+        return card != null && card.getCardName() != null && !card.getCardName().isEmpty();
+    }
+
```

Comment:
Introduced a validation method to make validating be a single responsibility, making the save method clearer and easier to test, improving readability, and simplifying future modifications.
https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html